src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.TryGet.cs

```diff
@@ -512,37 +512,24 @@ public bool TryGetGuid(out Guid value)
             }
 
             ReadOnlySpan<byte> span = HasValueSequence ? ValueSequence.ToArray() : ValueSpan;
-            int bytesConsumed = 0;
 
-            switch (span.Length)
+            if (_stringHasEscaping)
+            {
+                int idx = span.IndexOf(JsonConstants.BackSlash);
+                Debug.Assert(idx != -1);
+                span = JsonReaderHelper.GetUnescapedBytes(span, idx);
+            }
+
+            if (span.Length != 36 /* Valid Format D Guid length */)
```

Comment:
nit: Move this to a comment above the `if check`, rather than inline within the code.